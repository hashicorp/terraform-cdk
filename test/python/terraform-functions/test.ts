// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import { TestDriver } from "../../test-helper";

describe("python terraform functions test synth", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    driver = new TestDriver(__dirname);
    await driver.setupPythonProject();
  });

  test("synth generates JSON", async () => {
    await driver.synth();
    expect(
      driver.synthesizedStack("python-simple").toString()
    ).toMatchSnapshot();
  });
});
