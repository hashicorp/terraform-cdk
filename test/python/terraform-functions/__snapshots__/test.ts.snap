// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`python terraform functions test synth synth generates JSON 1`] = `
"{
  "//": {
    "metadata": {
      "backend": "local",
      "overrides": {
        "null_resource": [
          "triggers"
        ],
        "stack": [
          "terraform"
        ]
      },
      "stackName": "python-simple",
      "version": "stubbed"
    },
    "outputs": {
      "python-simple": {
        "computed": "computed"
      }
    }
  },
  "output": {
    "computed": {
      "value": "\${element(merge({\\"id\\" = null_resource.null-resource.id}, {\\"value\\" = \\"123\\"}), 1)}"
    }
  },
  "provider": {
    "null": [
      {}
    ]
  },
  "resource": {
    "null_resource": {
      "null-resource": {
        "//": {
          "metadata": {
            "path": "python-simple/null-resource",
            "uniqueId": "null-resource"
          }
        },
        "triggers": {
          "cluster_instance_ids": "foo"
        }
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "terraform.tfstate"
      },
      "remote": {
        "organization": "test",
        "workspaces": {
          "name": "test"
        }
      }
    },
    "required_providers": {
      "null": {
        "source": "hashicorp/null",
        "version": "3.1.1"
      }
    }
  }
}"
`;
